pycontrails/core/flight.py

```diff
@@ -2234,16 +2234,14 @@ def segment_rocd(
     if air_temperature is None:
         return out
 
-    else:
-        altitude_m = units.ft_to_m(altitude_ft)
-        T_isa = units.m_to_T_isa(altitude_m)
+    altitude_m = units.ft_to_m(altitude_ft)
+    T_isa = units.m_to_T_isa(altitude_m)
 
-        T_correction = np.empty_like(altitude_ft)
-        T_correction[:-1] = (0.5 * (air_temperature[:-1] + air_temperature[1:])) / (
-            0.5 * (T_isa[:-1] + T_isa[1:])
-        )
-        T_correction[-1] = np.nan
-        return T_correction * out
+    T_correction = np.empty_like(altitude_ft)
+    T_correction[:-1] = (air_temperature[:-1] + air_temperature[1:]) / (T_isa[:-1] + T_isa[1:])
```

Comment:
Besides un-indenting one level, I also removed the 0.5 from the numerator and denominator here -- it looked more readable to me.

Reply:
We could also use `np.diff` as well.

Reply:
The current implementation would be a hair faster. [`np.diff` does exactly](https://github.com/numpy/numpy/blob/893215af7f97105fa2dccb5bb560bc322dda6c77/numpy/lib/_function_base_impl.py#L1508-L1515) what we do here with a little more setup and tear down.